### PR TITLE
fix android build conflict

### PR DESCRIPTION
**Pull request by SunnyEver0**
#72 thanks

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 android {
-    compileSdkVersion 25
-    buildToolsVersion "25.0.2"
+    compileSdkVersion safeExtGet('compileSdkVersion', 25)
+    buildToolsVersion safeExtGet('buildToolsVersion', '25.0.2')
 
     defaultConfig {
-        minSdkVersion 16
-        targetSdkVersion 25
+        minSdkVersion safeExtGet('minSdkVersion', 16)
```
Comment (jordanbyron) [inline android/build.gradle]: This branch needs to be rebased to fix the merge conflict (new minSdkVersion)

Comment (SunnyEver0) [reply android/build.gradle]: @jordanbyron hello, I have resloved this conflict，thanks.

**Comment by jordanbyron** [comment]
@SunnyEver0 you never answered my question from above:

> Is there any documentation about libraries allowing their min and target sdk version from being overwritten by the container project? This seems brittle and potentially allows projects to build with this library despite it using a lower sdk version which is not supported.

**Comment by SunnyEver0** [comment]
@jordanbyron Hi, This change is to let the control of sdk be controlled by the project itself to avoid conflicts between the sdk compiled version and the plugin version of the project.Just like in[ react-native-vector-icons/android/build.gradle](https://github.com/oblador/react-native-vector-icons/blob/12ac7ecd9f3d353c43569c018af7ac31ae224d0a/android/build.gradle#L13)

**Comment by jordanbyron** [comment]
@SunnyEver0 thanks for linking to a project which uses this pattern. My concern about the current implementation dropping below our min supported SDK version still exists. Do you feel comfortable modifying your code so it will revert back to the min SDK version if the requested version is below our min? Something akin to this:

```
if rootProject.sdkVersion is present && rootProject.sdkVersion > ourMinSDK
  rootProject.sdkVersion
else
  ourMinSDK
end
```

**Comment by jordanbyron** [comment]
Closing due to inactivity. @SunnyEver0 you were on the right track here, but without the changes I requested I don't feel comfortable merging this in.

**Comment by SunnyEver0** [comment]
@jordanbyron hello, I am sorry for the comment that I have returned to you for so long.And In the case  if the requested version is below our min that you mentioned, I might think revert back to the min SDK version is unused.**Firstly, as @MBach said, there is no one will use sdk version lower than 21.Secondly, if the project sdk version lower than 21. they must set their sdk version bigger rather than the android built will be conflicted.**Thanks for reading my poor English😜

**Pull request closed without merging**